doc/overview.rst

```diff
@@ -985,6 +1037,88 @@ the
 
    predictor.predict(json.dumps(data).encode("utf-8"), {"ContentType": "application/list-text"})
 
+Built-in Components
+===================
+
+The following section provides information about the main components of built-in algorithms
+including pretrained models, model scripts, and model images.
+
+Pre-trained models
+------------------
+
+SageMaker maintains a model zoo of over 600 models from popular open source model hubs, such as
+TensorFlow Hub, Pytorch Hub, and HuggingFace. You can use the SageMaker Python SDK to fine-tune
+a model on your own dataset or deploy it directly to a SageMaker endpoint for inference.
+
+Model artifacts are stored as tarballs in an S3 bucket. Each model is versioned and contains a
+unique ID which can be used to retrieve the model URI. The following information describes the
+``model_id`` and ``model_version`` needed to retrieve the URI.
+
+.. container::
+
+   -  ``model_id``: A unique identifier for the JumpStart model.
+   -  ``model_version``: The version of the specifications for the
+      model. To use the latest version, enter ``"*"``. This is a
+      required parameter.
+
+To retrieve a model, first select a ``model ID`` and ``version`` from
+the :doc:`available models <./doc_utils/pretrainedmodels>`.
+
+.. code:: python
+
+   model_id, model_version = "huggingface-spc-bert-base-cased", "1.0.0"
+   scope = "training" # or "inference"
+
+Then use those values to retrieve the model as follows.
+
+.. code:: python
+
+   from sagemaker import model_uris
+
+   model_uri = model_uris.retrieve(
+       model_id=model_id, model_version=model_version, model_scope=scope
+   )
+
+Model scripts
+-------------
+
+To adapt pre-trained models for SageMaker, a custom script is needed to perform training
+or inference. SageMaker maintains a suite of scripts used for each of the models in the
+S3 bucket, which can be accessed using the SageMaker Python SDK Use the ``model_id`` and
+``version`` of the corresponding model to retrieve the related script as follows.
+
+.. code:: python
+
+   from sagemaker import script_uris
+
+   script_uri = script_uris.retrieve(
+       model_id=model_id, model_version=model_version, script_scope=scope
+   )
+
+Model images
+-------------
+
+A Docker image is required to perform training or inference on all
+SageMaker models. SageMaker relies on Docker images from the
+following repos https://github.com/aws/deep-learning-containers,
+https://github.com/aws/sagemaker-xgboost-container,
+and https://github.com/aws/sagemaker-scikit-learn-container. Use
+the ``model_id`` and ``version`` of the corresponding model to
+retrieve the related image as follows.
+
+.. code:: python
+
+   from sagemaker import image_uris
+
+   image_uri = image_uris.retrieve(
+       region=None,
+       framework=None,
+       image_scope=scope,
+       model_id=model_id,
+       model_version=model_version,
+       instance_type="ml.m5.xlarge",
```

Comment:
optional: since the image depends on the instance, how about first using the [instance_types.retrieve_default()](https://github.com/aws/sagemaker-python-sdk/blob/6a252f9ab16febdddb0c7b691677e6c0db41f826/src/sagemaker/instance_types.py#L26) utility, then pass it to the image utility?

Something like:
```py
from sagemaker import image_uris, instance_types

instance_type = instance_types.retrieve_default(
  model_id=model_id,
  model_version=model_version,
  scope=scope
)

image_uri = image_uris.retrieve(
       region=None,
       framework=None,
       image_scope=scope,
       model_id=model_id,
       model_version=model_version,
       instance_type=instance_type,
)
```

Reply:
+1